src/azure-cli-telemetry/azure/cli/telemetry/components/records_collection.py

```diff
@@ -40,7 +40,7 @@ def snapshot_and_read(self):
         for each in os.listdir(self._cache_dir):
             self._read_file(os.path.join(self._cache_dir, each))
 
-        shutil.rmtree(self._cache_dir,
+        shutil.rmtree(self._cache_dir,  # pylint: disable=deprecated-argument
```

Comment:
`onerror` is only deprecated in 3.12, but there is no equivalent in earlier versions: https://github.com/python/cpython/issues/112645. Disable the warning for now.

Ref: https://docs.python.org/3/library/shutil.html#shutil.rmtree